docs/trace_service_features.md

```diff
@@ -0,0 +1,128 @@
+# Trace Feature Coverage by Service
+
+## TOC
+
+- [Ad Service](#ad-service)
+- [Cart Service](#cart-service)
+- [Checkout Service](#checkout-service)
+- [Currency Service](#currency-service)
+- [Email Service](#email-service)
+- [Feature Flag Service](#feature-flag-service)
+- [Frontend Service](#frontend-service)
+- [Payment Service](#payment-service)
+- [Product Catalog Service](#product-catalog-service)
+- [Recommendation Service](#recommendation-service)
+- [Shipping Service](#shipping-service)
+
+## OpenTelemetry Feature Set by Services
+
+### Emoji Legend
+
+- Completed: :100:
+- Not Present (Yet): :construction:
+
+### [Ad Service](../src/adservice/README.md)
+
+tba
+
+### [Cart Service](../src/cartservice/README.md)
+
+tba
+
+### [Checkout Service](../src/checkoutservice/README.md)
+
+### Traces
+
+- :100: [Instrumentation
+  Libraries](https://opentelemetry.io/docs/concepts/instrumenting-library/)
+- :construction: [Manual Span
+  Creation](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/glossary.md#manual-instrumentation)
+- :construction: [Span Data
+  Enrichment](https://opentelemetry.io/docs/concepts/glossary/#metadata)
+- :construction: Interprocess Context Propagation
```

Comment:
I probably missed something here, what does "Interprocess Context Propagation" mean, and what's the difference compared to "Intra-service Context Propagation"?

Maybe the intention is:
* Context Propagation within a process
* Context Propagation across RPC boundaries

Reply:
Yes

Reply:
To double check given the approval is the preference to update to your language? I like the service framing but I could see the appeal of the more precise ipc / rpc etc.

Reply:
Here is my suggestion:
1. Given this PR is focusing on reorganizing the docs, we don't want to block the PR.
2. These terms (e.g. "Interprocess Context Propagation", "Intra-service Context Propagation" and "Trace Links") are not defined AND counter-intuitive (e.g. in computer world "Inter-process" is frequently used instead of "Interprocess", and is normally used under the context of IPC), we should revisit them in a separate PR. Ideally we should use defined terms and link to the definition (e.g. instead of using "Trace Links", I think we should use "Span Links" and add a link to https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/api.md#specifying-links). As a summary:
  2.1. We should always use defined terms whenever possible, and we should link to the definition.
  2.2. If 2.1 doesn't work, we may introduce/define our own terms, and the terms need to be well explained.